app/src/main/java/com/x8bit/bitwarden/data/platform/util/URIExtensions.kt

```diff
@@ -45,6 +45,27 @@ fun URI.parseDomainNameOrNull(resourceCacheManager: ResourceCacheManager): Domai
             )
         }
 
+/**
+ * Adds and HTTPS scheme to a valid URI if that URI has a valid host in the raw string but
+ * the standard parsing of `URI("foo.com")` is not able to determine a valid host.
+ * (i.e. val uri = URI("foo.bar:1090") -> uri.host == null)
+ */
+fun URI.addSchemeToUriIfNecessary(): URI {
```

Comment:
Great find! 

🤔 Assuming https means this will fail to match http hosts right? It's still common for lan services / local domains. If that's the case, could we instead change the match detection to only compare schema if it's present in our cipher? IIRC that's already the case for the port.

Reply:
@vvolkgang if the URL explicitly has an `http` scheme it will still match correctly, where this is being called it is specifically used to ensure the resulting URI is parsed correctly when attempting to grab out specific pieces: i.e. `uri.host` and `uri.port` specifically. Without a scheme this is not possible, so we are only adding the scheme if it does not already exist. If the user adds `http://foo.bar` or just `foo.bar` for an domain that is use `http` it will still match correctly. It will only add the `https` scheme if there is no other scheme present and it has a "host" or domain pattern based on the `.`. I am not seeing in the CipherMatching where we are comparing the schemes for the matching.

Reply:
I meant when the cipher is setup for a local service without a schema (e.g. `rpi.local`) and our app receives a request for `http://rpi.local`. If I understood correctly we would try to match `https://rpi.local` (from our cipher) to `http://rpi.local` (from the browser), would it fail in that case?

Reply:
@vvolkgang sorry for delay, been trying to track down one of the prod crashes, I tested this with an `http` site, with host matching, it is in fact working (video below). With the default domain matching it did not, but not because of the scheme. The host/domain of `testphp.vulnweb.com` being shortened to `vulnweb.com`, I believe this exposed another area where the URI is not being parsed as expected, adding this extension to our `String.getDomainOrNul()` call fixes this too.

https://github.com/user-attachments/assets/b59796d8-397c-46a2-ba5b-57ee530ead06

Reply:
Awesome, thanks for checking and improving that too!